xrmocap/core/estimation/mview_mperson_end2end_estimator.py

```diff
@@ -156,18 +257,163 @@ def run(
                 img_paths=img_paths,
                 video_paths=video_paths,
                 logger=self.logger)
-            n_view, n_frame, h, w, c = mview_batch_arr.shape
-            # TODO: infer keypoints3d here
-            kps3d_batch = None
-        keypoints3d_list = [
-            Keypoints(kps=kps3d_batch, mask=np.ones_like(kps3d_batch[..., 0]))
+            _, _, h, w, c = mview_batch_arr.shape
+
+            # prepare input data, img list and meta list
+            list_inputs = []
+            for _, img in enumerate(mview_batch_arr.squeeze()):
+                img_tensor = self.img_pipeline(img)
+                img_tensor = img_tensor.unsqueeze(0)
+                list_inputs.append(img_tensor)
+            meta = self.prepare_meta(cam_param, h, w)
+
+            frame_valid_pred_kps3d, _ = self.estimate_perception3d(
+                list_inputs, meta, self.inference_conf_thr)
+
+            pred_n_person = frame_valid_pred_kps3d.shape[0]
+            kps3d_batch[start_idx, :pred_n_person,
+                        ...] = frame_valid_pred_kps3d
+
+        # Convert array to keypoints instance
+        pred_keypoints3d = Keypoints(
+            dtype='numpy',
+            kps=kps3d_batch,
+            mask=kps3d_batch[..., -1] > 0,
+            convention=self.kps3d_convention,
+            logger=self.logger)
+
+        # Optimizing keypoints3d
+        pred_keypoints3d = self.optimize_keypoints3d(pred_keypoints3d)
+
+        # Fitting SMPL model
+        smpl_data_list = self.estimate_smpl(keypoints3d=pred_keypoints3d)
+
+        return pred_keypoints3d, smpl_data_list
+
+    def estimate_perception3d(self,
+                              img_arr: Union[None, np.ndarray],
+                              meta: Union[None, dict],
+                              threshold: float = 0.0):
+        """Estimate perception3d from images per frame.
+
+        Args:
+            img_arr (Union[None, np.ndarray]): _description_
+            meta (Union[None, dict]): _description_
+            threshold (float, optional): _description_. Defaults to 0.0.
```

Comment:
Add description.

Reply:
Fixed in [bda39aa](https://github.com/openxrlab/xrmocap/pull/46/commits/bda39aa6f75f5481a4372fde179ec7230f42e79c).